src/Cassandra/Requests/RequestHandler.cs

```diff
@@ -71,12 +71,12 @@ public RequestHandler(ISession session, Serializer serializer, IRequest request,
             Serializer = serializer;
             Statement = statement;
             Policies = _session.Cluster.Configuration.Policies;
-            RetryPolicy = session.Cluster.Configuration.Policies.RetryPolicy;
+            RetryPolicy = Policies.ExtendedRetryPolicy;
             if (statement != null)
             {
                 if (statement.RetryPolicy != null)
                 {
-                    RetryPolicy = statement.RetryPolicy;   
+                    RetryPolicy = statement.RetryPolicy.Wrap(Policies.ExtendedRetryPolicy);
```

Comment:
Minor suggestion, to avoid nested ifs. Could it be 
`if (statement != null && statement.RetryPolicy != null)`